### PR TITLE
Call Registration API in Trusts flow

### PR DESCRIPTION
**Pull request by HarrNM**

### CODE REVIEW

File: app/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnector.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.connectors
+
+import play.api.http.Status.{INTERNAL_SERVER_ERROR, OK}
+import play.api.libs.json.{JsObject, Json, Writes}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse, InternalServerException}
```
Comment (tvaidotas) [inline app/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnector.scala:21]:
```suggestion
import uk.gov.hmrc.http._
```

File: app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationController.scala [hunk 2]
```diff
@@ -20,7 +20,7 @@ import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.internalId
 import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
 import uk.gov.hmrc.http.InternalServerException
-import uk.gov.hmrc.minorentityidentificationfrontend.services.{BusinessVerificationService, JourneyService, StorageService}
+import uk.gov.hmrc.minorentityidentificationfrontend.services.{BusinessVerificationService, JourneyService, RegistrationOrchestrationService, StorageService}
```
Comment (tvaidotas) [inline app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationController.scala:23]:
```suggestion
import uk.gov.hmrc.minorentityidentificationfrontend.services._
```

File: it/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnectorISpec.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.connectors
+
+import play.api.test.Helpers.{INTERNAL_SERVER_ERROR, OK, await, defaultAwaitTimeout}
```
Comment (tvaidotas) [inline it/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnectorISpec.scala:19]:
```suggestion
import play.api.test.Helpers._
```

File: it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationControllerISpec.scala [hunk 4]
```diff
@@ -17,37 +17,27 @@
...
-import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.libs.json.Json
 import play.api.test.Helpers._
 import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants._
 import uk.gov.hmrc.minorentityidentificationfrontend.featureswitch.core.config.{BusinessVerificationStub, EnableFullTrustJourney, FeatureSwitching}
-import uk.gov.hmrc.minorentityidentificationfrontend.models.BusinessVerificationPass
-import uk.gov.hmrc.minorentityidentificationfrontend.stubs.{AuthStub, BusinessVerificationStub, StorageStub}
-import uk.gov.hmrc.minorentityidentificationfrontend.utils.{ComponentSpecHelper, WiremockHelper}
+import uk.gov.hmrc.minorentityidentificationfrontend.models.{BusinessVerificationPass, Registered}
+import uk.gov.hmrc.minorentityidentificationfrontend.stubs.{AuthStub, BusinessVerificationStub, RegisterStub, StorageStub}
```
Comment (tvaidotas) [inline it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationControllerISpec.scala:25]:
```suggestion
import uk.gov.hmrc.minorentityidentificationfrontend.stubs._
```

File: app/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnector.scala (hunk 1 above)
Comment (HarrNM) [reply app/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnector.scala]: Updated

File: app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationController.scala (hunk 2 above)
Comment (HarrNM) [reply app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationController.scala]: Updated

File: app/uk/gov/hmrc/minorentityidentificationfrontend/services/RegistrationOrchestrationService.scala [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.services
+
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.minorentityidentificationfrontend.connectors.RegistrationConnector
+import uk.gov.hmrc.minorentityidentificationfrontend.models.{BusinessVerificationPass, JourneyConfig, RegistrationNotCalled, RegistrationStatus}
```
[the remark this answers is not in the document]
Comment (HarrNM) [reply app/uk/gov/hmrc/minorentityidentificationfrontend/services/RegistrationOrchestrationService.scala]: Updated

File: it/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnectorISpec.scala (hunk 3 above)
Comment (HarrNM) [reply it/uk/gov/hmrc/minorentityidentificationfrontend/connectors/RegistrationConnectorISpec.scala]: Updated

File: it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationControllerISpec.scala (hunk 4 above)
Comment (HarrNM) [reply it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/BusinessVerificationControllerISpec.scala]: Updated

File: test/uk/gov/hmrc/minorentityidentificationfrontend/services/RegistrationOrchestrationServiceSpec.scala [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+package uk.gov.hmrc.minorentityidentificationfrontend.services
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.minorentityidentificationfrontend.connectors.mocks.MockRegistrationConnector
+import uk.gov.hmrc.minorentityidentificationfrontend.helpers.TestConstants._
+import uk.gov.hmrc.minorentityidentificationfrontend.httpparsers.StorageHttpParser.SuccessfullyStored
+import uk.gov.hmrc.minorentityidentificationfrontend.models.{BusinessVerificationFail, BusinessVerificationPass, Registered, RegistrationNotCalled}
```
[the remark this answers is not in the document]
Comment (HarrNM) [reply test/uk/gov/hmrc/minorentityidentificationfrontend/services/RegistrationOrchestrationServiceSpec.scala]: Updated

**Review by tvaidotas — APPROVED** [review]

**Pull request merged**